### PR TITLE
Use nim v1.6+

### PR DESCRIPTION
**Pull request by emizzle**
nim-stew now requires nim >= 1.6, so nim-leopard must be bumped

### CODE REVIEW

**Review by benbierens — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,7 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         cache_nonce: [ 1 ]
-        nim_version: [ 1.2.18, 1.4.8, 1.6.6 ]
+        nim_version: [ 1.6.6 ]
```
Comment (dryajov) [inline .github/workflows/test.yml:18]: Is 1.6.6 the latest nim we should be on?

Comment (dryajov) [reply .github/workflows/test.yml:18]: also, I think (I think!) if you just do `1.6` it should pick up the latest stuff.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,7 +10,7 @@ jobs:
       fail-fast: false
       matrix:
         cache_nonce: [ 1 ]
-        nim_version: [ 1.6.6 ]
+        nim_version: [ 1.6.20, stable ]
```
Comment (dryajov) [inline .github/workflows/test.yml:13]: Hmm, I don't know if we're ready for 2.0 yet? 

Also, I think the latest version in `nimbus-build-system` is 1.6.21, we should probably set it to that as well?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -145,7 +140,7 @@ jobs:
           if [[ ${{ matrix.platform.os }} = macos ]]; then
             export PATH="$(brew --prefix llvm@14)/bin:${PATH}"
             export LDFLAGS="-L$(brew --prefix libomp)/lib -L$(brew --prefix llvm@14)/lib -Wl,-rpath,$(brew --prefix llvm@14)/lib"
-            compiler_extra_options="-d:LeopardCmakeFlags='-DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=$(brew --prefix llvm@14)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm@14)/bin/clang++' -d:LeopardExtraCompilerlags='-fopenmp' -d:LeopardExtraLinkerFlags='-fopenmp -L$(brew --prefix libomp)/lib'"
+            compiler_extra_options="-d:LeopardCmakeFlags='-DENABLE_OPENMP=off -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=$(brew --prefix llvm@14)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm@14)/bin/clang++' -d:LeopardExtraCompilerlags='-fopenmp' -d:LeopardExtraLinkerFlags='-fopenmp -L$(brew --prefix libomp)/lib'"
```
Comment (dryajov) [inline .github/workflows/test.yml:143]: Interesting, why are we disabling opnemp?

Comment (emizzle) [reply .github/workflows/test.yml]: It was an attempt to get CI to build nim-leopard on macos arm, and disabling openmp was [suggested by csaba](https://github.com/status-im/leopard/pull/1#issuecomment-1146601769) as a way to get this to work.

Comment (emizzle) [reply .github/workflows/test.yml]: I've since removed this flag, as it didn't work

File: .github/workflows/test.yml (hunk 2 above)
Comment (emizzle) [reply .github/workflows/test.yml:13]: Aren't odd numbered patches considered unstable?

**Comment by emizzle** [comment]
Eventually, `macos-13` may become deprecated, and we may be forced to use an arm image for macos. In which case, using `asdf` instead of `choosenim` might be a simple way to support this: https://github.com/asdf-community/asdf-nim/blob/main/README.md#a-simple-example-using-github-actions

File: .github/workflows/test.yml (hunk 1 above)
Comment (emizzle) [reply .github/workflows/test.yml]: This workflow uses `choosenim`, which doesn't support the command `choosenim 1.6` as `1.6` isn't a version. `asdf` does support this command via `1.6:latest`, and one option is to use `asdf` if/when the `macos-13` (amd) runner is deprecated. Another option is `iffy`: https://github.com/codex-storage/asynctest/blob/main/.github/workflows/test.yml

**Comment by emizzle** [comment]
@dryajov are we ok to merge this?

File: .github/workflows/test.yml (hunk 3 above)
Comment (cskiraly) [reply .github/workflows/test.yml]: what do you mean by "it didn't work"?
On my system, as of now, leopard is building if I have `-DENABLE_OPENMP=off`.

If instead I have `-DENABLE_OPENMP=on`, it fails to build with this error:
```
[ 77%] Linking CXX executable bench_leopard
ld: Undefined symbols:
  ___kmpc_for_static_fini, referenced from:
      _.omp_outlined. in liblibleopard.a[3](LeopardCommon.cpp.o)
      _.omp_outlined. in liblibleopard.a[4](LeopardFF16.cpp.o)
      _.omp_outlined..2 in liblibleopard.a[4](LeopardFF16.cpp.o)
      _.omp_outlined..4 in liblibleopard.a[4](LeopardFF16.cpp.o)
      _.omp_outlined..6 in liblibleopard.a[4](LeopardFF16.cpp.o)
      _.omp_outlined..8 in liblibleopard.a[4](LeopardFF16.cpp.o)
      _.omp_outlined..10 in liblibleopard.a[4](LeopardFF16.cpp.o)
      ...
```

Maybe, following instructions in https://github.com/codex-storage/nim-leopard#openmp , it does link with OpenMP. Anyway, if we enable that as default, I think it should be a separate PR, and not part of a version bump PR.

Comment (cskiraly) [reply .github/workflows/test.yml]: OK, I see now. We were building with OpenMP in the past in the CI. Then you tried without it, and now you went back to original. All good (if it builds correctly) :-)

**Review by cskiraly — APPROVED** [review]
LGTM for squash merge.
If not squashed, would need a cleanup.

**Comment by cskiraly** [comment]
@emizzle One more thing: you should update the README, which has `Nim 1.2 or newer.`

File: .github/workflows/test.yml (hunk 3 above)
Comment (emizzle) [reply .github/workflows/test.yml]: > OK, I see now. We were building with OpenMP in the past in the CI (on amd64, not on arm64). Then you tried without it, hoping that would fix arm build, but it didn't. And now, you went back to original. All good (if it builds correctly) :-)

Yes. The CI runner changed from amd64 to arm64 (due to using the `macos-latest` runner). So I was trying to get leopard to compile using that runner (arm64). It didn't compile. Then I tried disabling openmp, which didn't help. So then I switched to using the `macos-13` runner which is amd64, and reverted the change to openmp.

> Maybe we would need a separate PR that actually enables arm64 build in the CI,

I agree, and I imagine that eventually the `macos-13` image will reach EOL and we will be forced to use arm64 a *some point*

**Pull request merged**